### PR TITLE
feat(frontend): add color switch

### LINKED ISSUE #1858: Possibilité de changer les couleurs pour avoir un meilleur contraste
Ajouter un lien "Accessibilité" dans le menu qui ouvre un menu déroulant : 
"Contraste : 
- standard
- renforcé"

Le click sur renforcé change les couleurs du site

### PR DESCRIPTION
**Pull request by UnbearableBear**
fix #1858
Cast an eye on the deployed branch
Screenshot 2019-12-18 at 14 28 04

### CODE REVIEW

File: packages/react-ui/src/field/Checkbox.js [hunk 1]
```diff
@@ -68,6 +68,14 @@ const StyledCheckbox = styled.input`
   & {
     visibility: hidden;
   }
+  /* stylelint-disable-next-line */
```
Comment (UnbearableBear) [inline packages/react-ui/src/field/Checkbox.js:71]: @nkrmr C'est là que j'ai viré le hover

Comment (nkrmr) [reply packages/react-ui/src/field/Checkbox.js:71]: OK top

File: packages/code-du-travail-frontend/pages/_app.js [hunk 2]
```diff
@@ -64,17 +66,21 @@ export default class MyApp extends App {
     const { Component, pageProps } = this.props;
     if (pageProps.statusCode) {
       return (
-        <ThemeProvider theme={theme.colors}>
-          <GlobalStyles />
-          <ErrorPage statusCode={pageProps.statusCode} />
+        <ThemeProvider>
+          <div>
```
Comment (lionelB) [inline packages/code-du-travail-frontend/pages/_app.js]: Fragment are not allowed as theme provider child ?

File: packages/code-du-travail-frontend/src/common/AccessibilityModal.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+} from "@socialgouv/react-ui";
+
+import { useTheme } from "../layout/ThemeProvider.js";
+
+export const AccessibilityModal = props => {
+  const [isModalOpened, setModalOpen] = useState(false);
+  const { currentTheme, toggleTheme } = useTheme();
+  return (
+    <>
+      {React.cloneElement(props.children, {
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/common/AccessibilityModal.js]: It not understood why you need to clone children in the first time. 
A render render prop approach would be simpler, less magic, and less error prone what if I do 
```
<AccessibilityModal onClick={trackAccessibility}>
   <Button>texte dans le bouton</Button>
</AccessibilityModal>
```

File: packages/code-du-travail-frontend/src/layout/ThemeProvider.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+const ThemeContext = React.createContext({
+  currentTheme: colors,
+  toggleTheme: () => {}
+});
+
+export const ThemeProvider = props => {
+  const [currentTheme, setCurrentTheme] = useState(colors);
+
+  useEffect(() => {
+    localStorage.getItem(HIGH_CONTRAST_STORAGE_KEY) === "true" &&
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/layout/ThemeProvider.js]: there is useLocalStorageHook :)

File: packages/react-ui/src/theme.js [hunk 5]
```diff
@@ -28,6 +44,14 @@ export const colors = {
 colors.primaryText = colors.white;
 colors.secondaryText = colors.white;
 
+export const highContrastColors = Object.entries(colors).reduce(
```
Comment (lionelB) [inline packages/react-ui/src/theme.js]: We could do the same using a level approach were use can choose is level 3 / 7 / 12

**Review by lionelB — CHANGES_REQUESTED** [review]
Cool, just few thinks about localStorage and the Modal approach

File: packages/code-du-travail-frontend/pages/_app.js (hunk 2 above)
Comment (UnbearableBear) [reply packages/code-du-travail-frontend/pages/_app.js]: The styled component ThemeProvider only accepts a single child and fragment is not accepted

File: packages/react-ui/src/theme.js (hunk 5 above)
Comment (UnbearableBear) [reply packages/react-ui/src/theme.js]: In fact I'm going to rename this one BlackAndWhiteTheme and next month we'll add a real highContrast mode (as discussed with clement). We already are in 3 so we only miss a 7 I guess but another time

File: packages/code-du-travail-frontend/src/layout/ThemeProvider.js (hunk 4 above)
Comment (UnbearableBear) [reply packages/code-du-travail-frontend/src/layout/ThemeProvider.js]: I'm going to take a look !

File: packages/code-du-travail-frontend/pages/_app.js (hunk 2 above)
Comment (lionelB) [reply packages/code-du-travail-frontend/pages/_app.js]: thanks !

File: packages/react-ui/src/theme.js (hunk 5 above)
Comment (lionelB) [reply packages/react-ui/src/theme.js]: Oh sorry, the comment was not meant to be actionnable now but more a thought for later

File: packages/code-du-travail-frontend/src/common/AccessibilityModal.js (hunk 3 above)
Comment (UnbearableBear) [reply packages/code-du-travail-frontend/src/common/AccessibilityModal.js]: To be honest I copy pasted the code from `ServiceDeRenseignementsModal` . Also, after taking a look at the modal usages in the project, we do indeed have a `renderProp` usage here:
```
                    <ConventionModal key="convention-modal">
                      {openModal => (
                        <Button variant="secondary" onClick={openModal}>
                          Chercher une convention collective
                        </Button>
                      )}
```
Is that ok if I refactor that the way above ?

Comment (lionelB) [reply packages/code-du-travail-frontend/src/common/AccessibilityModal.js]: yup !

File: packages/code-du-travail-frontend/pages/_app.js (hunk 2 above)
Comment (nkrmr) [reply packages/code-du-travail-frontend/pages/_app.js]: it's works like that for me on another project :

```
    <ThemeProvider theme={theme}>
      <>
        <GlobalStyle />
        <Routes />
      </>
    </ThemeProvider>
```

File: packages/code-du-travail-frontend/src/layout/ThemeProvider.js (hunk 4 above)
Comment (UnbearableBear) [reply packages/code-du-travail-frontend/src/layout/ThemeProvider.js]: Bon, je l'ai utilisé. Je ne suis pas sur d'avoir simplifié le code, au contraire, mais je l'ai utilisé !

**Review by lionelB — APPROVED** [review]

File: packages/code-du-travail-frontend/pages/_app.js (hunk 2 above)
Comment (UnbearableBear) [reply packages/code-du-travail-frontend/pages/_app.js]: Yep because in the end there is no html tag behind Routes and global styles

Comment (UnbearableBear) [reply packages/code-du-travail-frontend/pages/_app.js]: I'm wrong, tested

**Review by nkrmr — APPROVED** [review]

**Pull request merged**